geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/SimpleFeatureTypes.scala

```diff
@@ -431,6 +431,8 @@ object SimpleFeatureTypes {
     }
   }
 
+  // Do something smart here for keyword-userData??!?
```

Comment:
In general, we try to avoid comments like this, as they tend to get lost in the code. If it's something that you want to follow up on, create an issue in JIRA and reference it here as a TODO (if you don't have jira permissions someone can create it for you).
That said, it's probably easiest just to leave it as a string so that it will get persisted correctly, and split it in the RichSimpleFeatureType (like you're doing). Otherwise we will need to start adding converters on read/write for different user-data keys.